pkg/virt-controller/watch/vmi/vmi.go

```diff
@@ -1484,23 +1484,29 @@ func (c *Controller) deleteAllMatchingPods(vmi *virtv1.VirtualMachineInstance) e
 	vmiKey := controller.VirtualMachineInstanceKey(vmi)
 
 	for _, pod := range pods {
-		if pod.DeletionTimestamp != nil {
+		if !controller.IsControlledBy(pod, vmi) {
 			continue
 		}
 
-		if !controller.IsControlledBy(pod, vmi) {
+		if pod.DeletionTimestamp != nil && !isPodFinal(pod) {
 			continue
 		}
 
 		if err = c.deletePod(vmiKey, pod, v1.DeleteOptions{}); err != nil {
-			c.recorder.Eventf(vmi, k8sv1.EventTypeWarning, controller.FailedDeletePodReason, "Failed to delete virtual machine pod %s", pod.Name)
-			return err
+			if !k8serrors.IsNotFound(err) { // Skip the warning if the pod was already deleted, as this is an expected condition.
```

Comment:
It might be cleaner to not emit the `SuccessfulDelete` event again, when it was already deleted?